src/frontend/Semantic_error.ml

```diff
@@ -0,0 +1,552 @@
+open Core_kernel
+open Middle
+open Middle.Mir
+
+(** Type errors that may arise during semantic check *)
+module TypeError = struct
+  type t =
+    | MismatchedReturnTypes of returntype * returntype
+    | MismatchedArrayTypes
+    | InvalidRowVectorTypes
+    | IntExpected of string * unsizedtype
+    | IntOrRealExpected of string * unsizedtype
+    | IntIntArrayOrRangeExpected of unsizedtype
+    | IntOrRealContainerExpected of unsizedtype
+    | ArrayVectorRowVectorMatrixExpected of unsizedtype
+    | IllTypedAssignment of Ast.assignmentoperator * unsizedtype * unsizedtype
+    | IllTypedTernaryIf of unsizedtype * unsizedtype * unsizedtype
+    | ReturningFnExpectedNonReturningFound of string
+    | ReturningFnExpectedNonFnFound of string
+    | ReturningFnExpectedUndeclaredIdentFound of string
+    | NonReturningFnExpectedReturningFound of string
+    | NonReturningFnExpectedNonFnFound of string
+    | NonReturningFnExpectedUndeclaredIdentFound of string
+    | IllTypedStanLibFunctionApp of string * unsizedtype list
+    | IllTypedUserDefinedFunctionApp of
+        string
+        * (autodifftype * unsizedtype) list
+        * returntype
+        * unsizedtype list
+    | IllTypedBinaryOperator of operator * unsizedtype * unsizedtype
+    | IllTypedPrefixOperator of operator * unsizedtype
+    | IllTypedPostfixOperator of operator * unsizedtype
+    | NotIndexable of unsizedtype
+
+  (* let assignmentoperator_to_stan_math_fn = function
+    | Plus -> Some "assign_add"
+    | Minus -> Some "assign_subtract"
+    | Times -> Some "assign_multiply"
+    | Divide -> Some "assign_divide"
```

Comment:
What are these definitions doing here? Can they be deleted?

Reply:
Done